scipy/optimize/minpack.py

```diff
@@ -671,6 +672,16 @@ def curve_fit(f, xdata, ydata, p0=None, sigma=None, absolute_sigma=False,
     >>> plt.show()
 
     """
+
+    def check_len(z):
+        if len(z) == 0:
+            return False
+        else:
+            return True
```

Comment:
Could perhaps even write as a single line: `return len(z) != 0`

The `curve_fit` docstring suggests that `xdata` can be a `(k,M)` shaped array, so I'm not sure if there are weird corner cases where `len()` doesn't do what you want?

Reply:
i have changed it to a single line as suggested.